plugins/tokens/plugin.go

```diff
@@ -51,6 +51,60 @@ func createQueryHandler(mapper Mapper, queryPrefix string) app.AbciQueryHandler
 	return createAbciQueryHandler(mapper, queryPrefix)
 }
 
+const (
+	MaxUnlockItems = 10
+)
+
+func EndBlocker(ctx sdk.Context, timelockKeeper timelock.Keeper, swapKeeper swap.Keeper) {
+	if !sdk.IsUpgrade(sdk.SecondSunsetFork) {
+		return
+	}
+	logger := bnclog.With("module", "tokens")
+	logger.Info("unlock the time locks", "blockHeight", ctx.BlockHeight())
+
+	iterator := timelockKeeper.GetTimeLockRecordIterator(ctx)
+	defer iterator.Close()
+	i := 0
+	for ; iterator.Valid(); iterator.Next() {
+		if i >= MaxUnlockItems {
+			break
+		}
+		addr, id, err := timelock.ParseKeyRecord(iterator.Key())
+		if err != nil {
+			logger.Error("ParseKeyRecord error", "error", err)
+			continue
+		}
+		err = timelockKeeper.TimeUnlock(ctx, addr, id)
+		if err != nil {
+			logger.Error("TimeUnlock error", "error", err)
+			continue
+		}
+		i++
+	}
+
+	iterator = swapKeeper.GetSwapIterator(ctx)
+	defer iterator.Close()
```

Comment:
I will suggest you use a different iterator name, like:
```
swapiterator := swapKeeper.GetSwapIterator(ctx)
...
```

Because I am not sure what will happen if it calls `defer iterator.Close()` twice